### ISSUE TITLE
Transfer learning 

### ISSUE THREAD

**Issue by doit-ceo** [dbolya/yolact#454]
I'm trying to get transfer learning work, and followed every post I found here. I have 2 cans dataset (pepsi and lipton) and so far I did the following :

First all my images are box of 550px 

In config.py did the following changes 

```
COCO_CLASSES = ('lipton', 'pepsi')

COCO_LABEL_MAP = { 1:  1,  2:  2}
```
Then added my database to same file 

```
cans_dataset = dataset_base.copy({
    'name': 'cans',

    'train_images': '/content/drive/My Drive/yolactx/train/',
    'train_info':   '/content/drive/My Drive/yolactx/train/annotations.json',

    'valid_images': '/content/drive/My Drive/yolactx/test/',
    'valid_info':   '/content/drive/My Drive/yolactx/test/annotations.json',

    'has_gt': True,
    'class_names': ('lipton', 'pepsi')
})
```
And did the change to the database under yolact_base_config

```
# Dataset stuff
    'dataset': cans_dataset,
    'num_classes': len(cans_dataset.class_names) + 1,

```
Also modified the yolact.py to enable retraining 

 (line 490~)

```
        #self.load_state_dict(state_dict)
        try:
            self.load_state_dict(state_dict)
        except RuntimeError as e:
            print('Ignoring "' + str(e) + '"')
```
and  (line 630~)

```
#p = pred_layer(pred_x)
p = pred_layer(pred_x.detach())
```
I then run the training with 

`!python ./yolact/train.py --config=yolact_base_config --resume=/content/yolact/weights/yolact_plus_resnet50_54_800000.pth --start_iter=0`

Finally interference

`!python ./yolact/eval.py --trained_model=./yolact/weights/yolact_plus_resnet50_54_800000.pth --config=yolact_base_config --score_threshold=0.15 --top_k=15 --images=test_images:output_images
`

Now I'm getting really bad results and I think I made something really wrong ! 

![Selection_542](https://user-images.githubusercontent.com/8027780/82764581-79515b00-9e18-11ea-929a-cb4d457cc4ad.png)

**Comment by doit-ceo** [comment]
OK , I partially solved this, I found out that the new weights are stored in a different folder ! 

![Selection_544](https://user-images.githubusercontent.com/8027780/82796390-214d3f80-9e7e-11ea-85f1-a0b13a989b11.png)

So now I run the interference with 

```
!python ./yolact/eval.py --trained_model=./weights/yolact_base_1999_10000.pth --config=yolact_base_config --score_threshold=0.15 --top_k=15 --images=test_images:output_images
```

And much better results , but as you can see, still something wrong ?! 

![Selection_546](https://user-images.githubusercontent.com/8027780/82796700-a89ab300-9e7e-11ea-8f7c-a3d224e7456a.png)

**Comment by ic** [comment]
Just a couple comments:

* You are using Yolact++, which requires extra settings and dependencies. All good on that side?
* Transfer learning works best when the new task is a subset of the original task. It seems you are not in this situation, and should expect low performance without significant retraining. The backbone model you are using learned the original COCO classes. By redefining them, your first new class gets mixed with the first COCO class (person), at first and until enough training overrides the original learning. Please note label names are just names, the underlying class ID is where the confusion happens.
* Here I have no error on loading the backbone state dict. I think that is more than a warning. However I do not use Yolact++, just the plain Yolact.

**Comment by edwardnguyen1705** [comment]
Hi @doit-ceo ,
```!python ./yolact/train.py --config=yolact_base_config --resume=/content/yolact/weights/yolact_plus_resnet50_54_800000.pth --start_iter=0```

Your config file and weight file are not consistent.
